sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/cosmos/CosmosClientBuilderConfigurer.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.cosmos;
+
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.spring.identity.AbstractClientBuilderConfigurer;
+import com.azure.spring.identity.AzureKeyCredentialClientBuilderCustomizer;
+import com.azure.spring.identity.ClientBuilderCustomizer;
+import com.azure.spring.identity.DefaultSkipCredentialCallback;
+
+/**
+ * Configurer for extending Azure Cosmos service client builder configuration.
+ */
+public class CosmosClientBuilderConfigurer
+    extends AbstractClientBuilderConfigurer<ClientBuilderCustomizer<CosmosClientBuilder>, CosmosClientBuilder> {
+
+    private AzureKeyCredentialClientBuilderCustomizer<CosmosClientBuilder> azureKeyCredentialCustomizer;
+
+    public void setAzureKeyCredentialCustomizer(AzureKeyCredentialClientBuilderCustomizer<CosmosClientBuilder> azureKeyCredentialCustomizer) {
+        this.azureKeyCredentialCustomizer = azureKeyCredentialCustomizer;
+    }
+
+    @Override
+    public CosmosClientBuilder configure(CosmosClientBuilder builder) {
+        configureClientBuilder(builder);
+        if (azureKeyCredentialCustomizer != null) {
```

Comment:
How about omitting the DefaultSkipCredentialCallback and setting the credential by precedence, first setting the lower precedence and higher precedence.